### PR TITLE
Load mesos master URL from cook and use docker for integration tests

### PR DESCRIPTION
**Pull request by pschorf**

### CODE REVIEW

File: integration/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM python
```
Comment (shamsimam) [inline integration/Dockerfile]: Can we use `FROM python:3.5`?

File: integration/bin/build-docker-image.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (shamsimam) [inline integration/bin/build-docker-image.sh]: Please remove the black line.

File: integration/bin/build-docker-image.sh [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+#!/usr/bin/env bash
+
```
Comment (shamsimam) [inline integration/bin/build-docker-image.sh]: We should add a comment for usage.

File: integration/bin/run-integration.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
```
Comment (shamsimam) [inline integration/bin/run-integration.sh]: We should add a usage comment.

File: integration/tests/cook/util.py [hunk 5]
```diff
@@ -51,8 +51,16 @@ def retrieve_cook_url(varname='COOK_SCHEDULER_URL', value='http://localhost:1232
...
+def retrieve_mesos_url(varname='MESOS_PORT', value='5050'):
+    mesos_port = os.getenv(varname, value)
+    cook_url = retrieve_cook_url()
+    wait_for_cook(cook_url)
+    settings = session.get('%s/settings' % cook_url).json()
+    mesos_master_hosts = settings.get('mesos-master-hosts', ['localhost'])
+    resp = session.get('http://%s:%s/redirect' % (mesos_master_hosts[0], mesos_port), allow_redirects=False)
+    if resp.status_code != 307:
+        raise RuntimeError('Unable to find mesos leader, redirect endpoint returned %d' % resp.status_code)
+    mesos_url = 'http:%s' % resp.headers['Location']
```
Comment (shamsimam) [inline integration/tests/cook/util.py]: are we missing `//` after the `http:`?

File: integration/Dockerfile (hunk 1 above)
Comment (pschorf) [inline integration/Dockerfile]: I don't think we care about the minor version, why not just take the latest?

File: integration/bin/build-docker-image.sh (hunk 2 above)
Comment (pschorf) [inline integration/bin/build-docker-image.sh]: Done

File: integration/bin/build-docker-image.sh (hunk 3 above)
Comment (pschorf) [inline integration/bin/build-docker-image.sh]: Done

File: integration/bin/run-integration.sh (hunk 4 above)
Comment (pschorf) [inline integration/bin/run-integration.sh]: Done

File: integration/tests/cook/util.py (hunk 5 above)
Comment (pschorf) [inline integration/tests/cook/util.py]: the location header is formatted `//hostname:5050`

File: integration/tests/cook/util.py [hunk 6]
```diff
@@ -51,8 +51,16 @@ def retrieve_cook_url(varname='COOK_SCHEDULER_URL', value='http://localhost:1232
     return cook_url
 
 
-def retrieve_mesos_url(varname='MESOS_URL', value='http://localhost:5050'):
-    mesos_url = os.getenv(varname, value)
+def retrieve_mesos_url(varname='MESOS_PORT', value='5050'):
+    mesos_port = os.getenv(varname, value)
+    cook_url = retrieve_cook_url()
+    wait_for_cook(cook_url)
+    settings = session.get('%s/settings' % cook_url).json()
```
Comment (dposada) [inline integration/tests/cook/util.py]: There is a `settings` function in `test_basic.py`. Might be nice to move it into `util` and use it here.

Comment (pschorf) [inline integration/tests/cook/util.py]: Done

**Comment by pschorf** [comment]
Updated README

**Pull request merged**